platform/linuxbsd/display_server_x11.cpp

```diff
@@ -4991,7 +4991,13 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
 
 #ifdef DBUS_ENABLED
 	screensaver = memnew(FreeDesktopScreenSaver);
-	screen_set_keep_on(GLOBAL_DEF("display/window/energy_saving/keep_screen_on", true));
+
+	// Enforce default keep screen on value.
+	if (Engine::get_singleton()->is_editor_hint() || Engine::get_singleton()->is_project_manager_hint()) {
+		screen_set_keep_on(false);
+	} else {
+		screen_set_keep_on(GLOBAL_DEF("display/window/energy_saving/keep_screen_on", true));
```

Comment:
I don't think you need this check, instead you can add a `.editor` feature override to default it to false.
The `GLOBAL_DEF` should also be used only once to avoid multiple redefinitions (though here in practice it won't happen since it's different platforms), I'd move it to a common place (e.g. next to where other display/window properties are defined).

Reply:
Please also update the docs for this setting (in `doc/classes/ProjectSettings.xml` to make it clear which platforms support this setting).

Reply:
With this PR and #63900, it will work on all platforms except web, so "Works on desktop and mobile platforms." is correct and `ProjectSettings.xml` probably can be left as is.

Reply:
#63900 was merged so it should likely be amended too to handle the editor-specific disabling.

It might also be needed for Android, since it has editor support now too.

All in all, I'd suggest maybe keeping this part out of this PR (as well as my suggested refactoring for not defining the setting multiple times), and do the refactoring + editor handling in a follow-up once this one is merged.